components/brave_shields/browser/tracking_protection_service.cc

```diff
@@ -124,7 +124,7 @@ bool TrackingProtectionService::ShouldStoreState(HostContentSettingsMap* map,
                                                  int render_frame_id,
                                                  const GURL& top_origin_url,
                                                  const GURL& origin_url) const {
-  DCHECK_CURRENTLY_ON(BrowserThread::IO);
+  DCHECK_CURRENTLY_ON(BrowserThread::UI);
   if (!IsSmartTrackingProtectionEnabled()) {
```

Comment:
this probably also has unsafe access now because it was intended to be called on the IO thread